Platform/ElkhartlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -751,6 +751,7 @@ BoardInit (
     if (GetBootMode() != BOOT_ON_FLASH_UPDATE) {
       UpdatePayloadId ();
     }
+    GpioLockPads();

```

Comment:
A better place to call this would in GpioInit.c inside GpioConfigurePads() or GpioPadConfigTable() after the Gpio Programming is done?

Reply:
Yes, GpioLockPads() is supported in CommonSoc GpioLib. But I didnt call it from GpioInit.c in CommonSoc lib. Just because it is not clear yet if all projects want to lock their configs in Stage1/2 after Gpio programming.

Moving to CommonSoc GpioLib in general is a good idea , makes code much cleaner and we can get rid of redundant code. But I dont think it is going to help you with LockPads here. You still have to do it the way it is done in this patch-set.

Reply:
@stalamudupula if GpioLockPads called once either in end of GpioConfigurePads() or GpioPadConfigTable() then we need to unlock before configuring it? Or will it be better if we just lock it once all GPIO configuration is done?

Reply:
You are correct. SInce we are doing Gpio init multiple times, it is difficult to identify inside GpioInit.c when to lock. You can leave it as it is now. Thanks.

Reply:
@LeanSheng As per discussion i'll open another PR for the GPIO common lib porting and let's merge this patch for GpioLock fix.